packages/stream-tile-handler/src/tile-document-handler.ts

```diff
@@ -205,17 +213,61 @@ export class TileDocumentHandler implements StreamHandler<TileDocument> {
    * @param commitData - Commit to be verified
    * @param context - Ceramic context
    * @param controller - DID value
+   * @param streamId - Stream ID for the commit
    * @private
    */
   async _verifySignature(
     commitData: CommitData,
     context: Context,
-    controller: string
+    controller: string,
+    streamId: StreamID
   ): Promise<void> {
+    const cacao = await this._verifyCapability(commitData, context, streamId)
+
     await context.did.verifyJWS(commitData.envelope, {
       atTime: commitData.timestamp,
       issuer: controller,
       disableTimecheck: commitData.disableTimecheck,
+      capability: cacao,
     })
   }
+
+  /**
+   * Verifies capability attached to a signed commit
+   * @param commitData - Commit to be verified
+   * @param context - Ceramic context
+   * @param streamId - Stream ID for the commit
+   * @returns true if capability was present and was verified, false otherwise
+   */
+  async _verifyCapability(
+    commitData: CommitData,
+    context: Context,
+    streamId: StreamID
+  ): Promise<Cacao | undefined> {
+    const protectedHeader = commitData.envelope.signatures[0].protected
+    const decodedProtectedHeader = base64urlToJSON(protectedHeader)
+
+    if (decodedProtectedHeader.cap) {
+      const capIPFSUri = decodedProtectedHeader.cap
+      const capCID = CID.parse(capIPFSUri.replace('ipfs://', ''))
+      const cacao = (await context.ipfs.dag.get(capCID)).value as Cacao
+      const resources = cacao.p.resources as string[]
+      const payloadCID = commitData.cid.toString() // TODO: does this need to be a specific codec?
```

Comment:
Is this the CID of the JWS or the jws.link? (we should verify jws.link)

Reply:
btw, it does not need to be a specific codec.

Reply:
Because we can't know the CID of the DagJWS inside of the CACAO, since the hash of the Cacao is inside of the DagJWS header.